pkg/resource/model/custom_delta.go

```diff
@@ -33,7 +34,20 @@ func customSetDefaults(
 		}
 	}
 
-	if ackcompare.IsNil(a.ko.Spec.EnableNetworkIsolation) && ackcompare.IsNotNil(b.ko.Spec.EnableNetworkIsolation) {
-		a.ko.Spec.EnableNetworkIsolation = b.ko.Spec.EnableNetworkIsolation
+	// Default value of Mode is SingleModel
+	mode := aws.String("SingleModel")
+
+	if ackcompare.IsNotNil(a.ko.Spec.Containers) && ackcompare.IsNotNil(b.ko.Spec.Containers) {
+		for index := range a.ko.Spec.Containers {
+			if ackcompare.IsNil(a.ko.Spec.Containers[index].Mode) && ackcompare.IsNotNil(b.ko.Spec.Containers[index].Mode) {
```

Comment:
What if `len(a.ko.Spec.Containers) > len(b.ko.Spec.Containers)`?

Reply:
Ideally we want to just set these server side defaults regardless however for our current unit-test implementation we need to check b, ideally we wouldn't. Can be improved on further pr